bot/exts/filters/filtering.py

```diff
@@ -207,6 +208,12 @@ async def on_message_edit(self, before: Message, after: Message) -> None:
             delta = relativedelta(after.edited_at, before.edited_at).microseconds
         await self._filter_message(after, delta)
 
+    @Cog.listener()
+    async def on_voice_state_update(self, member: Member, before: VoiceState, after: VoiceState) -> None:
+        """Checks for bad words in usernames when users initially joins a voice channel."""
```

Comment:
Given it's `on_voice_state_update`, should we be more specific as to when this is triggered?
```suggestion
        """Checks for bad words in usernames when users join or switch voice channels."""
```

Reply:
With the condition being `if after.channel and not before.channel:`, it will only trigger when a member initially joins a voice channel. 

This is because we are checking the `after.channel` is not `None`, meaning they are now in a voice channel.
And we are also checking if `before.channel` was `None`, meaning they was not previously in a voice channel.

This condition isn't very easy to read, should I change it to the following for readability?

```python
if after.channel is not None and before.channel is None:
```

Reply:
> This condition isn't very easy to read, should I change it to the following for readability?
> 
> ```python
> if after.channel is not None and before.channel is None:
> ```

I personally don't really see this as an improvement. I actually find it better as-is.

> Additionally, I've been asked for it to only be triggered on initial joins.

To further expand on this: I requested this since there's no point in triggering when switching voice channels. In order to switch channels, one must first initially join a voice channel -- meaning that the moderators have already been alerted of any bad name. TL;DR; By not checking when switching channels, we prevents duplicate pings.

Reply:
The `check_bad_words_in_name` only pings once every 3 days, unless the docstring is misleading.

Reply:
> The `check_bad_words_in_name` only pings once every 3 days, unless the docstring is misleading.

Right. I didn't word my reasoning very well, but it still applies. It just saves a redundant function call.

Reply:
I am going to make this run on every single `voice_state_update` event, as we are already running this same function for every single message sent. So there shouldn't be any issues with slow down.

One thing we could do, if we are worried about that, is implementing an LRU cache on this [function](https://github.com/python-discord/bot/blob/2fc4f53bd7c3c669fc744ca307163d5c59eaa8a0/bot/exts/filters/filtering.py#L217). We would need to worry about invalidating the cache when the filters change. If we were to go with adding the LRU cache I would want to create a static function to be used for the LRU cache

Reply:
Current behaviour seems 👌, will mark this as resolved